### DISCUSSION TITLE
Anyway to identify the opened project

### DISCUSSION THREAD

**Discussion by andersstigel** [leolabs/ableton-js#91] [Q&A] [answered]
So my objective is to first make sure that the user has opened the project that I want them to.

Right now I am doing an spaghetti hack by checking that the names of the tracks are the ones I expect them to be. 

I guess the best thing, would be if there was some obvious unique identifier I could set in a project? And then check with ableton-js?

**Comment by leolabs** [comment, marked as answer]
You can set and read custom data using the `song.setData` and `song.getData` methods. This could allow you to give each project file a custom ID that you can check to ensure the correct file is opened.

Would that work for your use case?

**Reply by andersstigel** [reply]
Thanks @leolabs, 

For my use case it would work, if there is a easy way I could easily add the unique identifier to the project? 

I guess that using the song.setData and song.getData would require me to implement a new tool some kind of "add unique identifier to ableton live project" that i could run all my ableton live projects through before releasing them. And maybe that would be a bit much. What do you think? Any ideas?

**Reply by leolabs** [reply]
Other than that, I'm afraid there's no way to uniquely identify the current project using Live's Python API

**Reply by andersstigel** [reply]
Thanks, i'll look into creating a unique identifier setter tool 👍
